website/docs/sdk-reference/rust.mdx

```diff
@@ -0,0 +1,689 @@
+                "s": 0, // segment index, a valid index into the top-level segment array ("s")
+                "c": 1 // comparator, possible values: 0 -> IS IN SEGMENT, 1 -> IS NOT IN SEGMENT
+              }
+            }
+          ],
+          "s": { // alternatively, an array of Percentage Options ("p", see below) can also be specified
+            "v": { // the value served when the rule is selected during evaluation
+              "b": true
+            },
+            "i": "bcfb84a7"
+          }
+        }
+      ],
+      "p": [ // array of Percentage Options
+        {
+          "p": 10, // % value
+          "v": { // the value served when the Percentage Option is selected during evaluation
+            "b": true
+          },
+          "i": "bcfb84a7"
+        },
+        {
+          "p": 90,
+          "v": {
+            "b": false
+          },
+          "i": "bddac6ae"
+        }
+      ],
+      "v": { // fallback value, served when none of the Targeting Rules match,
+             // no Percentage Options are defined or evaluation of these is not possible
+        "b": false // depending on the setting type, another type of value may need to be specified:
+                   // text setting -> "s": string
+                   // whole number setting -> "i": number
+                   // decimal number setting -> "d": number
+      },
+      "i": "430bded3" // variation id (for analytical purposes)
+    }
+  }
+}
+```
+
+For a more comprehensive specification of the config JSON v6 format, you may refer to [this JSON schema document](https://github.com/configcat/config-json/blob/main/V6/config.schema.json).
+
+### Map
+
+You can set up the SDK to load your feature flag & setting overrides from a `HashMap<String, configcat::Value>`.
+
+```rust
+use configcat::{Client, MapDataSource, OverrideBehavior, Value};
+
+let map: MapDataSource = [
+    ("enabledFeature", Value::Bool(true)),
+    ("disabledFeature", Value::Bool(false)),
+    ("intSetting", Value::Int(5)),
+    ("doubleSetting", Value::Float(1.2)),
+    ("stringSetting", Value::String("test".to_owned())),
+].into();
+
+let client = Client::builder("localhost")
+    // highlight-next-line
+    .overrides(Box::new(map), OverrideBehavior::LocalOnly)
+    .build()
+    .unwrap();
+```
+
+## `get_all_keys()`
+
+You can query the keys of each feature flag and setting with the `get_all_keys()` method.
+
+```rust
+let client = Client::new("#YOUR-SDK-KEY#").unwrap();
+let keys = client.get_all_keys().await;
+```
+
+## `get_all_values()`
+
+Evaluates and returns the values of all feature flags and settings. Passing a [User Object](#user-object) is optional.
+
+```rust
+let client = Client::new("#YOUR-SDK-KEY#").unwrap();
+let values = await client.get_all_values(None).await;
+
+// invoke with User Object
+let user = User::new("#UNIQUE-USER-IDENTIFIER#");
+let values_with_user = await client.get_all_values(Some(user)).await;
+```
+
+## `get_all_details()`
```

Comment:
I think this is called get_all_value_details in the other SDKs. Maybe it is worth changing it.

Reply:
Done